### PR TITLE
feat(conventional_commits): Add exclamation on commit title for breaking change

### PR DESCRIPTION
**Pull request by bastien31**
Add a configuration (disable by default) to allow exclamation in commit title to ease identify BC simple git log parsing.

## Description
Conventional commit allow to put breaking change exclamation character in git commit title.
This is a useful convention to identify BC commit from the title only.
As it is supported in commitizen parsing, add the option to generate it.

## Checklist

- [x] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [x] Add test cases to all the changes you introduce
- [x] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [x] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [x] Ensure backward compatibility is maintained
  - [x] Document any manual testing steps performed
- [x] Update the documentation for the changes

### Documentation Changes

- [x] Run `poetry doc` locally to ensure the documentation pages renders correctly
- [x] Check and fix any broken links (internal or external) in the documentation

> When running `poetry doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

### CODE REVIEW

**Comment by bastien31** [comment]
Tested manually:
image
image

**Comment by bastien31** [comment]
After a rebase on master, I had to add following fix [186fbd83fc7d2fa8c363fa2a08a7d901a657517c ](https://github.com/commitizen-tools/commitizen/pull/1576/commits/186fbd83fc7d2fa8c363fa2a08a7d901a657517c) to get tests passed.
Also, on my side, I need to add "deprecated" library to get commitizen running.

**Comment by Lee-W** [comment]
yep, this needs https://github.com/commitizen-tools/commitizen/pull/1591 to be merged

File: commitizen/cz/conventional_commits/conventional_commits.py [hunk 1]
```diff
@@ -154,11 +154,21 @@ def message(self, answers: ConventionalCommitsAnswers) -> str:  # type: ignore[o
         footer = answers["footer"]
         is_breaking_change = answers["is_breaking_change"]
 
+        # Check if breaking change exclamation in title is enabled
+        breaking_change_exclamation_in_title = self.config.settings.get(
+            "breaking_change_exclamation_in_title", False
+        )
```
Comment (bearomorphism) [inline commitizen/cz/conventional_commits/conventional_commits.py]: How about

```py
title = f"{prefix}{scope}"
if is_breaking_change and self.config.settings.get("breaking_change_exclamation_in_title"):
    title += "!"
```

File: docs/config.md [hunk 2]
```diff
@@ -96,6 +96,15 @@ Use annotated tags instead of lightweight tags. [See difference][annotated-tags-
 
 Create custom commit message. Useful to skip CI. [Read more][bump_message]
 
+### `breaking_change_exclamation_in_title`
+
+Type: `bool`
+
+Default: `false`
```
Comment (bearomorphism) [inline docs/config.md]: nit

```
Default: `False`
```

File: docs/config.md [hunk 3]
```diff
@@ -96,6 +96,15 @@ Use annotated tags instead of lightweight tags. [See difference][annotated-tags-
...
+
+When true, breaking changes will be indicated by an exclamation mark in the commit title (e.g., `feat!: breaking change`).
+When false, breaking changes will be indicated by `BREAKING CHANGE:` in the footer. [Read more][breaking-change-exclamation]
```
Comment (bearomorphism) [inline docs/config.md]: Just noticed that there are some inconsistencies between this line and your change.

Users might expect that the string `BREAKING CHANGE: ` will not be added into the footer when `breaking_change_exclamation_in_title` is `True`.

Additionally, could you help to check if the link "Read more" works? If not, you can add the link at the end of this document.

File: commitizen/cz/conventional_commits/conventional_commits.py (hunk 1 above)
Comment (bastien31) [reply commitizen/cz/conventional_commits/conventional_commits.py]: Ok for the simplification, I would suggest something a little bit different to keep footer handling.

File: docs/config.md (hunk 3 above)
Comment (bastien31) [reply docs/config.md]: Good catch the link was not working and was quite stupid (ref to mylself), I did fix it and fwd to the proper destination.

**Comment by bastien31** [comment]
Ready for review.

**Review by woile — APPROVED** [review]

**Review by Lee-W — APPROVED** [review]
